Counterexamples/VitaliSetNotMeasurable.lean

```diff
@@ -0,0 +1,318 @@
+    have eq : vS.r x' (vRep ⟦x'⟧) := by
+      refine Quotient.eq.mp ?_
+      symm
+      apply vRep_spec
+    simp only [Setoid.r] at eq
+    simpa [← h_x2]
+  have h_xy2 : x - y ∈ Icc (-1) 1 := by
+    simp only [mem_Icc] at h_x1 h_y1 ⊢
+    constructor <;> linarith
+  simp only [vitaliUnion, image_add_right, mem_iUnion, mem_preimage, exists_prop]
+  use (x - y)
+  constructor
+  · rw [vI, mem_inter_iff]
+    constructor <;> assumption
+  · simpa [vitaliSet']
+
+/-- Therefore, the volume of vitaliUnion is between 1 and 3 (inclusive). -/
+lemma vitaliUnion_volume_range : 1 ≤ volume vitaliUnion ∧ volume vitaliUnion ≤ 3 := by
+  have h1 : volume (Icc (0 : ℝ) 1) = 1 := by simp [volume_Icc]
+  have h2 : volume (Icc (-1 : ℝ) 2) = 3 := by simp [volume_Icc] ; norm_num
+  constructor
+  · rw [← h1]
+    exact volume_mono vitaliUnion_lower_bound
+  · rw [← h2]
+    exact volume_mono vitaliUnion_upper_bound
+
+/-- The shifted copies of vitaliSet in vitaliUnion are pairwise disjoint. -/
+lemma vitali_pairwise_disjoint : vI.PairwiseDisjoint vitaliSet' := by
+  intro x x_vI y y_vI x_ne_y
+  refine Set.disjoint_iff.mpr ?_
+  intro z
+  simp only [mem_inter_iff, mem_empty_iff_false, imp_false, not_and]
+  intro z_x z_y
+  simp only [vitaliSet', vitaliSet, image_add_right, preimage_setOf_eq, mem_setOf_eq] at z_x
+  simp only [vitaliSet', vitaliSet, image_add_right, preimage_setOf_eq, mem_setOf_eq] at z_y
+  have ⟨t_x, rep_x⟩ := z_x
+  have ⟨t_y, rep_y⟩ := z_y
+  have vRep_eq : vS.r (vRep t_x) (vRep t_y) := by
+    simp only [vS, mem_range]
+    simp only [rep_x, rep_y, add_sub_add_left_eq_sub, sub_neg_eq_add]
+    have : x ∈ range ((↑) : ℚ → ℝ) := by exact mem_of_mem_inter_right x_vI
+    have ⟨q_x, q_x_eq⟩ := mem_range.mp (mem_of_mem_inter_right x_vI)
+    have ⟨q_y, q_y_eq⟩ := mem_range.mp (mem_of_mem_inter_right y_vI)
+    use (-q_x + q_y)
+    simp [← q_x_eq, ← q_y_eq]
+  have vT_eq := Quotient.sound vRep_eq
+  simp only [vRep_spec] at vT_eq
+  have x_eq_y : x = y := by
+    calc x = z - ↑(vRep t_x) := by { simp [rep_x] }
+         _ = z - ↑(vRep t_y) := by { simp [vT_eq] }
+         _ = y := by { simp [rep_y] }
+  contradiction
+
+/-- vI is a countable set. -/
+lemma vI_countable : vI.Countable := by
+  refine Countable.mono inter_subset_right ?_
+  apply countable_range
+
+/-- This is the main result that will lead to a contradiction:
+if the vitaliSet is null-measurable, then the volume of vitaliUnion is
+the sum of countably many copies of vitaliSet.  -/
+lemma vitaliUnion_volume_sum (hm : NullMeasurableSet vitaliSet volume) :
+    volume vitaliUnion = ∑' (_ : ↑vI), volume vitaliSet := by
+  have hm' : ∀ i ∈ vI, NullMeasurableSet (vitaliSet' i) volume := by
+    intro i i_vI
+    rw [vitaliSet']
+    apply shift_nullmeasurable hm
+  rw [vitaliUnion, biUnion_volume vI_countable vitali_pairwise_disjoint hm']
+  refine tsum_congr ?_
+  intro i
+  rw [vitaliSet', shift_volume]
+
+/-- vI is an infinite set. -/
+lemma vI_infinite : vI.Infinite := by
```

Comment:
This should follow more easily from [`Set.Icc_infinite`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Interval/Set/Infinite.html#Set.Icc_infinite).